src/Entity/Int32EntityId.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+use InvalidArgumentException;
+
+/**
+ * Interface for EntityIds that can be converted to a unique positive 32 bit integer in the range
+ * [1..2147483647], and back from the entity type and the number. Entity types that do not meet this
+ * criteria should not implement this interface.
+ *
+ * Entity types are not required and not guaranteed to implement this interface. Use the full string
+ * serialization whenever you can and avoid using numeric IDs.
+ *
+ * @since 6.1
+ *
+ * @license GPL-2.0+
+ * @author Thiemo Mättig
+ */
+interface Int32EntityId {
+
+	/**
+	 * @since 6.1
+	 *
+	 * @param string $idSerialization
+	 *
+	 * @throws InvalidArgumentException When the provided serialization does not allow getNumericId
+	 *  to return a unique positive 32 bit integer.
+	 */
+	public function __construct( $idSerialization );
```

Comment:
Seeing a constructor in an interface is odd to me - it's essentially a static function, right? And when we call a constructor, we should always know the concrete class... 
Specifying this here means that no class that implements this can have any additional constructor arguments, either. What do we need that guarantee/restricting for?